hibernate-core/src/main/java/org/hibernate/type/descriptor/jdbc/UuidAsBinaryJdbcType.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * SPDX-License-Identifier: LGPL-2.1-or-later
+ * Copyright Red Hat Inc. and Hibernate Authors
+ */
+package org.hibernate.type.descriptor.jdbc;
+
+import org.hibernate.type.descriptor.ValueExtractor;
+import org.hibernate.type.descriptor.WrapperOptions;
+import org.hibernate.type.descriptor.java.JavaType;
+
+import java.sql.CallableStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Types;
+import java.util.Arrays;
+
+import static org.hibernate.type.SqlTypes.UUID;
+
+/**
+ * @author Jan Schatteman
+ */
+public class UuidAsBinaryJdbcType extends BinaryJdbcType {
+
+	public static final UuidAsBinaryJdbcType INSTANCE = new UuidAsBinaryJdbcType();
+
+	@Override
+	public int getDdlTypeCode() {
+		return Types.VARBINARY;
```

Comment:
I understand that the bug only emerges when using VARBINARY, but the correct type here is BINARY:

```suggestion
		return Types.BINARY;
```